yjit/src/yjit.rs

```diff
@@ -45,6 +45,14 @@ pub extern "C" fn rb_yjit_init(yjit_enabled: bool) {
 
     // If --yjit-disable, yjit_init() will not be called until RubyVM::YJIT.enable.
     if yjit_enabled {
+
+    // Call YJIT hooks before enabling YJIT to avoid compiling the hooks themselves.
+    // Needs to be done without VM lock held because it can context switch.
+    unsafe {
+        let yjit = rb_const_get(rb_cRubyVM, rust_str_to_id("YJIT"));
+        rb_funcall(yjit, rust_str_to_id("call_jit_hooks"), 0);
+    }
```

Comment:
```suggestion
        // Call YJIT hooks before enabling YJIT to avoid compiling the hooks themselves.
        // Needs to be done without VM lock held because it can context switch.
        unsafe {
            let yjit = rb_const_get(rb_cRubyVM, rust_str_to_id("YJIT"));
            rb_funcall(yjit, rust_str_to_id("call_jit_hooks"), 0);
        }
```